plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -137,15 +197,20 @@ func zpool() ([]string, error) {
 	return run("zpool", []string{"list", "-Hp", "-o", "name,health,size,alloc,free,fragmentation,capacity,dedupratio"}...)
 }
 
+func zdataset() ([]string, error) {
```

Comment:
How about getting the information to gather as a parameter? You could then synchronize with the `gatherDatasetStats()` function and do not keep the names in two different locations...